pkgs/test_core/lib/src/runner.dart

```diff
@@ -299,10 +293,20 @@ class Runner {
             return false;
           }
 
-          // Skip tests that don't start on `line` or `col` if specified.
-          var line = suite.config.line;
-          var col = suite.config.col;
-          if (line != null || col != null) {
+          final testSelections = suite.config.testSelections;
+          if (testSelections!.isEmpty) {
```

Comment:
nit:

```suggestion
          final testSelections = suite.config.testSelections!;
          if (testSelections.isEmpty) {
```

It also feels a bit weird that this is nullable, should it have a default (empty) instead?

Reply:
We are not consistent with how we handle `null` and empty collections in our configuration.

Reply:
The reason this one seems like it should be non-null is because we are requiring it to be set here, it feels a bit brittle (we are relying on a call to `selectTests` before it is valid). We could also handle null here.

Reply:
As written `null` and empty do mean different things, especially when merging.

I think, though, that `{}` can share a meaning with `{TestSelection()}` as written, so I can take a look at moving `{}` to hold the meaning that `null` does currently.

Reply:
I considered writing code to enforce that you can't `selectTests` more than once. This `!` does enforce that `selectTests` was called at least once.

Reply:
Made it non-null.
I added some asserts around having it be set to non-empty only once.